### PR TITLE
Add category handling for grouping of requirement information

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #16176)

### CODE REVIEW

File: framework/doc/content/css/alert_moose.css [hunk 1]
```diff
@@ -123,9 +102,16 @@
 }
 
 .moose-alert-title{
-    padding-top:6px;
-    padding-bottom:6px;
+    padding-top:2px;
+    padding-bottom:2px;
 }
```
Comment (loganharbour) [inline framework/doc/content/css/alert_moose.css:107]: Decide you didn't like the title?

File: python/MooseDocs/extensions/sqa.py [hunk 2]
```diff
@@ -678,10 +750,8 @@ class RenderSQARequirementDesign(autolink.RenderLinkBase):
     def findDesign(self, filename, design, line):
         node = self.translator.findPage(design, throw_on_zero=False)
         if node is None:
-            msg = "Unable to locate the design page: {}\n    {}:{}" \
-                  .format(design, filename, line)
-            LOG.error(msg)
-            return None
+            msg = "Unable to locate the design page: {}\n    {}:{}"
+            raise exceptions.MooseDocsException(msg, design, filename, line)
```
Comment (loganharbour) [inline python/MooseDocs/extensions/sqa.py:754]: 👍🏼

File: python/MooseDocs/test/common/demo [hunk 3]
```diff
@@ -24,16 +25,19 @@
     requirement = "Requirement Group One"
     verification = bibtex.md
     [group0-a]
+       type = Deanne
        detail = 1D
     []
     [group0-b]
+       type = Deanne
```
Comment (loganharbour) [inline python/MooseDocs/test/common/demo:32]: 👋🏼 

Glad to see Deanne making her way into MOOSE.

**Review by loganharbour — APPROVED** [review]
I like this simplification now that we're getting closer to hitting the requirements everywhere 👍🏼

File: framework/doc/content/css/alert_moose.css (hunk 1 above)
Comment (aeslaughter) [reply framework/doc/content/css/alert_moose.css:107]: Yup, it was taking up too much space.

**Pull request merged**